edx_notifications/stores/mongo/store_provider.py

```diff
@@ -0,0 +1,258 @@
+import copy
+import datetime
+import pymongo
+from pymongo import MongoClient
+import pytz
+from edx_notifications import const
+from edx_notifications.exceptions import BulkOperationTooLarge
+from edx_notifications.stores.mongo.mongo_models import MongoUserNotification
+from edx_notifications.stores.sql.store_provider import SQLNotificationStoreProvider
+
+
+class MongoNotificationStoreProvider(SQLNotificationStoreProvider):
+    """
+
+    """
+
+    def __init__(self, **kwargs):
+        super(MongoNotificationStoreProvider, self).__init__(**kwargs)
+        self.client = MongoClient(kwargs.get('host'), kwargs.get('port'))
+        self.db = self.client[kwargs.get('database_name')]
+        self.collection = self.db.user_notification
+        # self.bulk = self.collection.initializeUnorderedBulkOp()
```

Comment:
See: http://api.mongodb.org/python/2.7.2/api/pymongo/collection.html#pymongo.collection.Collection.ensure_index

For pymongo v. 2.7.2 there is an ensure_index() that a client can use to (quickly) test to see if an index is available, if not, then create it.